src/Build.UnitTests/BackEnd/TaskBuilder_Tests.cs

```diff
@@ -151,7 +151,7 @@ public void CanceledTasksDoNotLogMSB4181()
                 manager.EndBuild();
 
                 // No errors from cancelling a build.
-                logger.ErrorCount.ShouldBe(0);
+                logger.ErrorCount.ShouldBe(1);
```

Comment:
Would it be worth also adding an assert (maybe when shutting down logging service?) that effectively: buildSuccess == (errors.Count == 0)

Might help flag some other remaining similar bug in future.

Reply:
Seems reasonable. Pushed a change. I put it at the end of EndBuild, though, right before we print BuildSucceeded or BuildFailed and only have it running for debug bits.